[ENG-3803] docs: add InstallIntegration wizard mode (beta) documentation (#554)
amp-labs/docs: src/embeddable-ui-components.mdx

![](/images/1f151f2-Example_for_docs_1.png) 

> ### Wizard mode (Beta)

Reviewer: This should be #### (so it's nested underneath `Install integration`, and not at the same level )

Reviewer: can you also remove the headings that's one level nested? (e.g. How to activate, Behavior, etc.) If you reduce the length of this section (by using gif, and not individual screenshots), then the subheadings won't be necessary. 

The content map on the right side of the page is too long otherwise, and this isn't important enough of a section to have so much real estate in the content map. 
<img width="397" height="802" alt="Screenshot 2026-04-07 at 10 16 58 PM" src="https://github.com/user-attachments/assets/0fae3567-19e2-49ef-909c-3c8757bb3660" />
se.

> <Warning>The wizard mode is currently in beta preview. The `variant="wizard"` prop is hidden and does not appear in the public TypeScript declarations as of `@amp-labs/react` v2.13.1. Please see [supported features](#supported-features) for details on what is currently available.</Warning>

Reviewer: move the warning below the next paragraph. it's better to explain the feature and then give the warning, than the other way around.

> The wizard mode replaces the standard "all-at-once" configuration view with a guided, multi-step installation flow. It walks your end users through connecting, selecting objects, configuring fields/mappings, and reviewing before creating an installation.

Reviewer: ```suggestion
The wizard mode replaces the standard configuration view with a guided, multi-step installation flow during initial set up. It walks your end users through connecting, selecting objects, configuring fields/mappings, and reviewing before creating an installation.
```

#### How to activate

Pass `variant="wizard"` to the `InstallIntegration` component:

```JSX
<InstallIntegration
  integration={myIntegrationName}
  consumerRef={userId}
  consumerName={userFullName}
  groupRef={teamId}
  groupName={teamName}
  // @ts-expect-error -- needed since hidden as an internal prop
  variant="wizard"
/>
```

#### Behavior

- **New installations**: When no installation exists, the wizard renders.
- **Existing installations**: The standard configuration UI is shown instead.

> #### The 5-step flow

Reviewer: I actually meant that you just attach a gif (e.g. the one in https://ampersand.slab.com/posts/install-wizard-demo-document-vpdm9911#hzzev-demo), otherwise this page is indeed getting too long.

Reviewer: also can you put the gif above "how to activate"? then you are showing off the feature first and then telling people how to get access to it.

Author: moved giph up.

**Step 1: Connect**

The user authenticates with the integration provider via OAuth.

![](/images/wizard-step-1-connect.png)

**Step 2: Select Objects**

The user picks which objects to sync from the provider. A "Write Enabled" toggle appears when the object is selected and supported via the deployed `amp.yaml`.

![](/images/wizard-step-2-select-objects.png)

**Step 3: Configure Objects**

Per-object configuration with a tabbed multi-object interface.

For each selected object, the user walks through sub-pages. Users can click back to a previously configured object.

![](/images/wizard-step-3-configure-objects.png)

**Step 4: Review & Create**

A summary screen showing all configured objects:
The user clicks "Create Installation" to submit. Error messages are shown inline if submission fails.

![](/images/wizard-step-4-review.png)

**Step 5: Success**

A confirmation screen.

#### Supported features

| Feature | Supported |
|---|---|
| Read objects | Yes |
| Write objects | Yes |
| Required fields | Yes |
| Optional fields | Yes |
| Field mappings | Yes |
| Object mappings | Yes |

> | Value mappings | No |

Reviewer: for the unsupported features, can you please add links so the reader can actually understand what is not being supported?

Author: added links
